FRAuth/FRAuth/Callbacks/DeviceSigningVerifierCallback.swift

```diff
@@ -0,0 +1,243 @@
+        guard let callbackType = json[CBConstants.type] as? String else {
+            throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
+        
+        guard let outputs = json[CBConstants.output] as? [[String: Any]], let inputs = json[CBConstants.input] as? [[String: Any]] else {
+            throw AuthError.invalidCallbackResponse(String(describing: json))
+        }
+        
+        // parse outputs
+        var outputDictionary = [String: Any]()
+        for output in outputs {
+            guard let outputName = output[CBConstants.name] as? String, let outputValue = output[CBConstants.value] else {
+                throw AuthError.invalidCallbackResponse("Failed to parse output")
+            }
+            outputDictionary[outputName] = outputValue
+        }
+        
+        self.userId = outputDictionary[CBConstants.userId] as? String
+        
+        guard let challenge = outputDictionary[CBConstants.challenge] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing challenge")
+        }
+        self.challenge = challenge
+        
+        guard let title = outputDictionary[CBConstants.title] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing title")
+        }
+        self.title = title
+        
+        guard let subtitle = outputDictionary[CBConstants.subtitle] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing subtitle")
+        }
+        self.subtitle = subtitle
+        
+        guard let promptDescription = outputDictionary[CBConstants.description] as? String else {
+            throw AuthError.invalidCallbackResponse("Missing description")
+        }
+        self.promptDescription = promptDescription
+        
+        self.timeout = outputDictionary[CBConstants.timeout] as? Int
+        
+        //parse inputs
+        var inputNames = [String]()
+        for input in inputs {
+            guard let inputName = input[CBConstants.name] as? String else {
+                throw AuthError.invalidCallbackResponse("Failed to parse input")
+            }
+            inputNames.append(inputName)
+        }
+        
+        guard let jwsKey = inputNames.filter({ $0.contains(CBConstants.jws) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing jwsKey")
+        }
+        self.jwsKey = jwsKey
+        
+        guard let clientErrorKey = inputNames.filter({ $0.contains(CBConstants.clientError) }).first else {
+            throw AuthError.invalidCallbackResponse("Missing clientErrorKey")
+        }
+        self.clientErrorKey = clientErrorKey
+        
+        try super.init(json: json)
+        type = callbackType
+        response = json
+    }
+    
+    
+    /// Sign the device.
+    /// - Parameter completion Completion block for Device binding result callback
+    open func sign(completion: @escaping DeviceBindingResultCallback) {
```

Comment:
Should this be DeviceSigningResultCallback?

Reply:
Created DeviceSigningResultCallback and used that in DeviceSigningVerifierCallback